catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternFormulaParser.java

```diff
@@ -298,6 +306,49 @@ private FormulaElement collision() throws InternFormulaParserException {
 		return lookTree;
 	}
 
+	private FormulaElement distanceToTouchPosition() throws InternFormulaParserException {
+		String currentSpriteName = ProjectManager.getInstance().getCurrentSprite().getName();
+		int spriteCount = 0;
+
+		for (Sprite sprite : ProjectManager.getInstance().getSceneToPlay().getSpriteList()) {
+			if (sprite.getName().compareTo(currentSpriteName) == 0) {
+				spriteCount++;
+			}
+		}
+
+		if (!(spriteCount == 1)) {
+			throw new InternFormulaParserException("Parse Error, Sprite was not found");
+		}
```

Comment:
Always keep [premature optimization](https://en.wikipedia.org/wiki/Program_optimization#When_to_optimize) in mind.

However, you hinted me to something else.
IMO it doesn't make sense to search for the `CurrentSprite` in the `SceneToPlay`* anyway. A lot of code builds on the premise that the opened Scene,Sprite in the UI is the `getCurrentScene`,`getCurrentSprite`.
Searching for the second sprite however makes sense. If you've deleted the second sprite, the formula becomes invalid.**
What do you think?

\* `SceneToPlay == CurrentScene`. I'll create a PR for that.
\** I'm not 100% sure if *a parser* should check this, but it's not important for this PR.